File: 0005-longest-palindromic-substring/0005-longest-palindromic-substring.java
class Solution {
    public String longestPalindrome(String s) {
        String ans="";
         int p=s.length();
        int start,end,arambam=0,antham=Integer.MIN_VALUE;
        for(int i=0;i<p;i++){
            start=i;
            end=i;
            while(start>=0 && end<p && s.charAt(start)==s.charAt(end)){
                if(end-start+1>antham){
                    antham=end-start+1;
                    ans=s.substring(start,end+1);
                    
                }
                start--;
                end++;
            }
            
            start=i;
            end=i+1;
            while(start>=0 && end<p && s.charAt(start)==s.charAt(end)){
                if(end-start+1>antham){
                   antham=end-start+1;
                   ans=s.substring(start,end+1);
                }
                start--;
                end++;
            }
            
            
        }
        return ans;
        
    }
}

File: 0005-longest-palindromic-substring/0005-longest-palindromic-substring.cpp
class Solution {
public:
    string longestPalindrome(string s) {
        int siz=s.size();
        int arambam=0;
        int antham=1;
        int start,end=0;
        for(int i=0;i<siz;i++){
            start=i-1;
            end=i;
            while(start>=0 && end<siz && s[start]==s[end]){
                if(end-start+1>antham){
                    arambam=start;
                    antham=end-start+1;
                }
                start--;
                end++;
            }
             start=i-1;
            end=i+1;
            while(start>=0 && end<siz && s[start]==s[end]){
                if(end-start+1>antham){
                    arambam=start;
                    antham=end-start+1;
                }
                start--;
                end++;
            }
            
        }
        return s.substr(arambam,antham);
        
        
    }
};